### PR TITLE
SWIFT_MODULE_DEPENDS_DARWIN

### PR DESCRIPTION
**Pull request by compnerd**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @jrose-apple @shahmishal @Rostepher @gottesmm

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by gottesmm** [comment]
@compnerd since these are automatically updated, I imagine whatever thing is used to update them must be updated as well. @Rostepher @zisko: how is this working now?

**Comment by compnerd** [comment]
@gottesmm yeah, I was hoping that one of you guys would know how that is being handled.

**Comment by jrose-apple** [comment]
The updates are computed by utils/find-overlay-dependencies.sh and friends, though there are some additional complexities around Apple-internal only content that, well, stinks for us at Apple.

I'm going to abstain from this one and let Ross and Jacob decide whether it's a good change.

**Comment by compnerd** [comment]
@jrose-apple the second change in this set is something that is more of a clean up, and if it ends up being painful for the internal only content, then we can drop it.  Saw the opportunity, so whipped up a patch :-)

**Comment by compnerd** [comment]
CC: @Rostepher @zisko

**Comment by compnerd** [comment]
@gottesmm - anyone other than @zisko and @Rostepher that can decide what we want to do with this?

File: cmake/modules/AddSwift.cmake [hunk 1]
```diff
@@ -1387,15 +1390,62 @@ endfunction()
 #   Sources to add into this library.
 function(add_swift_library name)
   set(SWIFTLIB_options
-      SHARED STATIC OBJECT_LIBRARY IS_STDLIB IS_STDLIB_CORE IS_SDK_OVERLAY
-      TARGET_LIBRARY FORCE_BUILD_FOR_HOST_SDK
-      API_NOTES_NON_OVERLAY DONT_EMBED_BITCODE HAS_SWIFT_CONTENT
-      FORCE_BUILD_OPTIMIZED NOSWIFTRT)
+        API_NOTES_NON_OVERLAY
```
Comment (Rostepher) [inline cmake/modules/AddSwift.cmake]: I don't think we should use this PR as a means to re-format these arguments. I'd much prefer if we kept whitespace changes in separate, non-functional PRs if possible. Makes it not only easier to review, but also distills PRs such as this one down to just the functional changes.

File: cmake/modules/AddSwift.cmake [hunk 2]
```diff
@@ -1387,15 +1390,62 @@ endfunction()
...
+        SWIFT_MODULE_DEPENDS_TVOS
+        SWIFT_MODULE_DEPENDS_WATCHOS
+        TARGET_SDKS)
+
   cmake_parse_arguments(SWIFTLIB
-    "${SWIFTLIB_options}"
-    "INSTALL_IN_COMPONENT;DEPLOYMENT_VERSION_OSX;DEPLOYMENT_VERSION_IOS;DEPLOYMENT_VERSION_TVOS;DEPLOYMENT_VERSION_WATCHOS"
-    "DEPENDS;LINK_LIBRARIES;SWIFT_MODULE_DEPENDS;SWIFT_MODULE_DEPENDS_OSX;SWIFT_MODULE_DEPENDS_IOS;SWIFT_MODULE_DEPENDS_TVOS;SWIFT_MODULE_DEPENDS_WATCHOS;SWIFT_MODULE_DEPENDS_FREEBSD;SWIFT_MODULE_DEPENDS_LINUX;SWIFT_MODULE_DEPENDS_CYGWIN;SWIFT_MODULE_DEPENDS_HAIKU;FRAMEWORK_DEPENDS;FRAMEWORK_DEPENDS_WEAK;FRAMEWORK_DEPENDS_OSX;FRAMEWORK_DEPENDS_IOS_TVOS;LLVM_COMPONENT_DEPENDS;FILE_DEPENDS;TARGET_SDKS;C_COMPILE_FLAGS;SWIFT_COMPILE_FLAGS;SWIFT_COMPILE_FLAGS_OSX;SWIFT_COMPILE_FLAGS_IOS;SWIFT_COMPILE_FLAGS_TVOS;SWIFT_COMPILE_FLAGS_WATCHOS;LINK_FLAGS;PRIVATE_LINK_LIBRARIES;INTERFACE_LINK_LIBRARIES;INCORPORATE_OBJECT_LIBRARIES;INCORPORATE_OBJECT_LIBRARIES_SHARED_ONLY"
-    ${ARGN})
+                        "${SWIFTLIB_options}"
```
Comment (Rostepher) [inline cmake/modules/AddSwift.cmake]: Standard CMake styling does not require that arguments are "visually" indented (like in Python) to the same column as the first argument. In fact, the standard style in this project is all subsequent lines are to be indented 2 spaces. These sorts of whitespace changes and re-stylings add a lot of diff churn and potential merge conflicts.

File: stdlib/public/SDK/CoreAudio/CMakeLists.txt [hunk 3]
```diff
@@ -1,21 +1,38 @@
 cmake_minimum_required(VERSION 3.4.3)
 include("../../../../cmake/modules/StandaloneOverlay.cmake")
 
-add_swift_library(swiftCoreAudio ${SWIFT_SDK_OVERLAY_LIBRARY_BUILD_TYPES} IS_SDK_OVERLAY
-  CoreAudio.swift
-  ../../../public/core/WriteBackMutableSlice.swift
+add_swift_library(swiftCoreAudio
+                    ${SWIFT_SDK_OVERLAY_LIBRARY_BUILD_TYPES}
```
Comment (Rostepher) [inline stdlib/public/SDK/CoreAudio/CMakeLists.txt]: Same here, we should indent 2 spaces on subsequent lines.

**Review by Rostepher — CHANGES_REQUESTED** [review]
This sort of functional change should not also include all these whitespace/formatting changes. In particular the indentation in the various overlays is not in keeping with the standard formatting style.

Can we move those changes to separate "gardening" PRs and slim this one down? And as @jrose-apple mentioned, these arguments are automagically updated by a few scripts in the `utils` directory. It's likely those will also need to be updated to know about these shared dependencies.

**Comment by compnerd** [comment]
@Rostepher - yeah, I want to figure out if this is something that you think is helpful/valuable or not.  This needs to be rebased which will remove all the other stuff around it - the commits were broken up accordingly to that anways.

The idea here is that you are unifying the commonality of the iOS/tvOS/watchOS/macOS dependency tree.  There are so many cases where you find checks to see if the build is of that type and then perform a behaviour.  It just seems that this simplifies some of that logic.

**Comment by compnerd** [comment]
@Rostepher - ping?

**Comment by compnerd** [comment]
Seems that there is no interest from Apple on this ...

**Pull request closed without merging**